src/pages/en/reference/directives-reference.md

```diff
@@ -67,6 +67,25 @@ const cmsContent = await fetchHTMLFromMyCMS();
 <Fragment set:html={cmsContent}>
 ```
 
+`set:html={Promise<string>}` injects an HTML string into an element that is wrapped in a Promise.
+
+This can be used to inject HTML stored externally, such as a database.
+
+```astro
+---
+import api from '../db/api.js';
+---
+<article set:html={api.getArticle(Astro.props.id)}></article>
+```
+
+`set:html={Promise<Response>}` injects a [Response](https://developer.mozilla.org/en-US/docs/Web/API/Response) into an element.
```

Comment:
What exactly gets injected here? The full contents of `await response.text()`? Or do we do something more to that content, like only what’s in `<body>` or something for fetched HTML?

Reply:
The full contents. It's not `await response.text()` though, it streams in the contents.